webapp/src/Entity/Problem.php

```diff
@@ -287,26 +283,82 @@ public function getSpecialCompareArgs(): ?string
         return $this->special_compare_args;
     }
 
-    public function setCombinedRunCompare(bool $combinedRunCompare): Problem
+    public function setTypes(array $types): Problem
     {
-        $this->combined_run_compare = $combinedRunCompare;
+        $types = array_unique($types);
+
+        $this->types = 0;
+        foreach ($types as $type) {
+            if ($type === 'pass-fail') {
+                $this->types |= self::TYPE_PASS_FAIL;
+            } elseif ($type === 'scoring') {
+                $this->types |= self::TYPE_SCORING;
+            } elseif ($type === 'multi-pass') {
+                $this->types |= self::TYPE_MULTI_PASS;
+            } elseif ($type === 'interactive') {
+                $this->types |= self::TYPE_INTERACTIVE;
+            }
+        }
         return $this;
     }
 
-    public function getCombinedRunCompare(): bool
+    public function getTypes(): array
+    {
+        $types = [];
+        if ($this->isPassFailProblem()) {
+            $types[] = 'pass-fail';
+        }
+        if ($this->isScoringProblem()) {
+            $types[] = 'scoring';
+        }
+        if ($this->isMultipassProblem()) {
+            $types[] = 'multi-pass';
+        }
+        if ($this->isInteractiveProblem()) {
+            $types[] = 'interactive';
+        }
```

Comment:
Why not do this the same as `getTypesForForm` below? What's the difference between these except for returning an array of strings or constants? This could easily be mapped from one to the other.

Reply:
yeah, that is much better, see new code